metaphor/custom/query_attributions/config.py

```
@@ -0,0 +1,36 @@
+from dataclasses import field as dataclass_field
+from typing import Dict, List
+
+from pydantic.dataclasses import dataclass
+
+from metaphor.common.base_extractor import BaseConfig
+from metaphor.common.dataclass import ConnectorConfig
+from metaphor.common.utils import is_email
+from metaphor.models.metadata_change_event import (
+    DataPlatform,
+    QueryAttribution,
+    QueryAttributions,
+)
+
+
+@dataclass(config=ConnectorConfig)
+class PlatformQueryAttributions:
+    platform: DataPlatform = DataPlatform.UNKNOWN
+    queries: Dict[str, str] = dataclass_field(default_factory=lambda: dict())
+
+    def to_mce_query_attributions(self) -> QueryAttributions:
+        return QueryAttributions(
+            platform=self.platform,
+            queries=[
+                QueryAttribution(query_id=k, user_email=v)
+                for k, v in self.queries.items()
+                if is_email(v)
```

Ideally we want to error out for invalid email addresses instead of dropping them silently. Mabye adding it as a validator to the model?

---

sure.